src/hooks/useActiveLocale.ts

```diff
@@ -14,6 +16,22 @@ function parseLocale(maybeSupportedLocale: string): SupportedLocale | undefined
   )
 }
 
+/**
+ * Returns the supported locale read from the url query parameter (lng=)
+ */
+function urlLocale(): SupportedLocale | undefined {
+  const search = location.search || location.hash.match(/\?.*/)?.[0]
```

Comment:
`location.hash.match(/\?.*/)?.[0]` is necessary in case the query parameter is placed after a hash. IDK if this is spec-compliant, but it is definitely done by our current codebase, so it must be supported.